cmd/podman/containers/create.go

```diff
@@ -299,12 +299,31 @@ func CreateInit(c *cobra.Command, vals entities.ContainerCreateOptions, isInfra
 			}
 			vals.Env = env
 		}
-		if c.Flag("cgroups").Changed && vals.CgroupsMode == "split" && registry.IsRemote() {
+
+		cgroupsChanged := c.Flag("cgroups").Changed
+		if cgroupsChanged && vals.CgroupsMode == "split" && registry.IsRemote() {
 			return vals, fmt.Errorf("the option --cgroups=%q is not supported in remote mode", vals.CgroupsMode)
 		}
 
-		if c.Flag("pod").Changed && !strings.HasPrefix(c.Flag("pod").Value.String(), "new:") && c.Flag("userns").Changed {
-			return vals, errors.New("--userns and --pod cannot be set together")
+		if c.Flag("userns").Changed {
```

Comment:
I wonder if we should add this to Libpod's `container_validate.go` to make sure that, even if we add additional flags, setting a user namespace while joining a pod is not allowed

Reply:
I have no idea how to distinguish normal containers and infra containers at libpod level. The latter can have userns set.

Reply:
`ctr.config.IsInfra` bool should do it

Reply:
I have just changed it to a warning in CLI.